src/main/java/com/google/devtools/build/lib/bazel/repository/downloader/ProxyHelper.java

```diff
@@ -51,6 +51,19 @@ public ProxyHelper(Map<String, String> env) {
    */
   public Proxy createProxyIfNeeded(URL requestedUrl) throws IOException {
     String proxyAddress = null;
+    String noProxyUrl = env.get("no_proxy");
+    if (Strings.isNullOrEmpty(noProxyUrl)) {
+      noProxyUrl = env.get("NO_PROXY");
+    }
+    if (!Strings.isNullOrEmpty(noProxyUrl)) {
+      String[] noProxyUrlArray = noProxyUrl.split(",");
+      String requestedHost = requestedUrl.getHost();
+      for (int i = 0; i < noProxyUrlArray.length; i++) {
+        if(requestedHost.endsWith(noProxyUrlArray[i])) {
```

Comment:
What if NO_PROXY contains "bazel.build" and the hostname is "this-is-not-bazel.build"? Doesn't it accidentally match the hostname then? One way to fix it might be:

```java
if (noProxyUrlArray[i].startsWith(".")) {
  // This entry applies to sub-domains only.
  if (requestedHost.endsWith(noProxyUrlArray[i])) {
    return Proxy.NO_PROXY;
  }
} else {
  // This entry applies to the literal hostname and sub-domains.
  if (requestedHost.equals(noProxyUrlArray[i])
      || requestedHost.endsWith("." + noProxyUrlArray[i])) {
    return Proxy.NO_PROXY;
  }
}
```

What do you think?

Reply:
Yea that's fair, let me look around how other tools are supporting this.  This seems like a good fix though